ecl/hql/hqlutil.cpp

```diff
@@ -5709,7 +5711,8 @@ IHqlExpression * extractCppBodyAttrs(unsigned lenBuffer, const char * buffer)
                 }
             }
         }
-        prev = buffer[i];
+        if (!ignore)
+            prev = next;
```

Comment:
When the current position in buffer is updated, 'next' would not reflect the updated position.  This means for example, certain input text will be incorrectly handled:
  // one line comment 
  *x = 3; 
  /* second comment */

With this code, the entire 3 lines would be commented out.